### ISSUE TITLE
`no-unused-expressions` with do syntax

### ISSUE THREAD

**Issue by hzoo** [babel/eslint-plugin-babel#13]
``` js
const a = 1;
const b = do {
  a; // lint error here
}
```

Ref https://github.com/babel/babel-eslint/issues/158 @jacobp100

**Comment by Cap32** [comment]
+1

**Comment by frankhinek** [comment]
Any chance this might ever be fixed?

**Comment by fritx** [comment]
+1 related: https://github.com/eslint/eslint/issues/6310#issuecomment-223276106

**Comment by alexrussell** [comment]
Out of interest, is this not being done because `do` is stage-0 or is it simply a general lack of dev time to focus here?

**Comment by hzoo** [comment]
You'd be right 😁 there's always a lack of dev time for all oss projects including this one - being stage-0 also means less priority as well. There's no one specifically working on all of these tasks so it's more waiting for a pr - it's just a place for us to store rules before they give to stage-4 and put into ESLint

**Comment by jscinoz** [comment]
For what it's worth, `do` is [stage-1 now](https://github.com/tc39/proposals).

**Comment by valmassoi** [comment]
I'm just doing
```javascript
const a = 1;
const b = do { /*eslint-disable */
  a;
} /*eslint-enable */
```
For now 🙁

**Comment by kentcdodds** [comment]
Anyone wanna work on this? I don't personally use `do` expressions yet, so I have no reason to work on this myself. But if it's bothering you, you're welcome to [open a PR](http://makeapullrequest.com/) to fix it :)

**Comment by duongdev** [comment]
Anyone fixed this?

**Comment by bradennapier** [comment]
:( also breaks auto formatting

**Comment by lyleunderwood** [comment]
I put together an initial PR for this, feedback appreciated: https://github.com/babel/eslint-plugin-babel/pull/131

**Comment by jscinoz** [comment]
`do` can also erroneously trigger `no-new`:

```javascript
const testClassInstance = do {
  const TestClass = someFunctionCreatingAnES6Class();

  new TestClass(); // eslint: error Do not use 'new' for side effects  no-new
}
```

**Comment by lednhatkhanh** [comment]
Hope to see this being merged soon.

**Comment by lyleunderwood** [comment]
@lednhatkhanh this has been merged actually, it's available in 5.1.0.

**Comment by ljharb** [comment]
Closed by #131.
